### ISSUE TITLE
Don't hardcode "wheel" group in tmpfiles.d

### ISSUE THREAD

**Issue by martinpitt** [systemd/systemd#2492]
There is no `wheel` group on Debian and derivatives. This makes `systemd-tmpfiles-setup` fail due to the ACLs in `tmpfiles.d/systemd.conf.m4` like

```
a+ /run/log/journal/%m - - - - d:group:adm:r-x,d:group:wheel:r-x
```

So far we have dropped those with a downstream patch. But as discussed on last Friday's sprint, we want to run the downstream integration autopkgtests on upstream PRs -- for that it's impractical to apply additional patches as they often fail due to fuzziness. This is one of just two simple issues that currently prevent that, the other is PR #2491.

There's some bikeshedding to do for this, in order of descending preference from my POV:
1. auto-detect if `wheel` exists on the build host, and use it iff so
2. `--enable-journal-wheel-group` (default: detect if it exists on the build host)
3. `--with-journal-acl-groups=adm,wheel` 

As `wheel` is (should be?) a static system group and thus always exist or not within the same distribution, IMHO option 1 is the simplest one. IMHO it does not make that much sense to allow fine-grained configuration here. Opinions?

**Comment by poettering** [comment]
We use "adm" as group for users who shall "see" what root can see, and "wheel" as group for users who shall be able to "do" what root can do. We also define it in sysusers.d/basic.conf.

Wouldn't it be better to just create the group on Debian machines, if it currently doesn't know it? I mean, you create the "systemd-journal" group, too, so why not at the "wheel" group?

**Comment by martinpitt** [comment]
> Wouldn't it be better to just create the group on Debian machines

There is already the `admin` group for that, which can run `sudo` by default, and gets admin-like privs in polkit. Creating yet another group for that purpose would be too confusing.

**Comment by johannbg** [comment]
Afaikt Debian and Debian based distributions seem to be the odd ball here by not creating/using the wheel group so it must be confusing for all administrators coming from other linux distributions and OS like *BSD, OS-X, Solaris etc which expect the group to exist there no? + Did not Debian rename that group to sudo at one point and root at another etc which causes even more confusion? 
What's the technical reason for Debian and Debian based distribution not creating the wheel group by default in the first place?

**Comment by vcaputo** [comment]
At risk of getting drawn into a bikeshed argument, wheel was never a good name for this group, even after reading its entry in the jargon file one is left scratching their head in bewilderment.

**Comment by johannbg** [comment]
That name got inherited from the Tenex system from the person who posses a wheel bit to be able to unwedge the hung tape drives anyway as good or bad the naming may be in the end the naming is irrelevant. 

What matters is common practice which makes a good upstream default and the common practice happens to be the "wheel" group in this particular case and there has to be some technical reason why Debian is not creating it by default since not doing so goes against common practice. 
( surely it cannot be due to some stallman-ism or bad naming )

**Comment by martinpitt** [comment]
> There is already the admin group for that

Sorry, I meant `sudo`. Ubuntu started out with `admin`, but has been deprecated in favor of Debian's `sudo` group which got added many years later. (It can't ever remove it of course, but it stopped documenting it and setting it up for new installs).

`wheel` isn't on http://refspecs.linuxbase.org/LSB_4.1.0/LSB-Core-generic/LSB-Core-generic/usernames.html, and seriously, I'm not going down the path of trying to justify history -- it is what it is. `wheel` is certainly not any clearer or "better" in any way (in fact, it has no connotation to "Administrator" at all). Merely creating that group doesn't serve any purpose, it's just confusing and cruft.

**Comment by johannbg** [comment]
You should refer to the current LSB standard ( 5.0 ) anyway according the spec ( which relevance I'm unsure of since it seems to be never up to speed with what's happening in the linux ecosystem and common practice and Debian is not LSB compliant distribution and never will be afaik ) root is the only required username and group and the rest is legacy or optional ( including adm )

**Comment by poettering** [comment]
i'd be fine with a configure switch --with-wheel-group= or so...

**Comment by johannbg** [comment]
Why not drop it completely like you did with the lock group? What's gained with an added switch for a single distro and it's derivatives?

**Comment by mbiebl** [comment]
Dropping the concept of a wheel aka. sudo group won't work. So +1 for a --with-wheel-group= switch.

**Comment by mbiebl** [comment]
fwiw, I would phrase it more neutrally, like --with-admin-group=, since that captures the meaning better.

**Comment by johannbg** [comment]
Calling it "--with-admin-group" will make it conflict with the actual adm group and it makes no sense to add a switch just for the wheel group either. You either add a ( individual ) switch for _all_ groups and users since root is the only required user and group ( by lsb anyway ) or none not just those a single distro and it's derivatives might have chosen to rename to something else ( or in the case of Fedora invent ).

**Comment by martinpitt** [comment]
Michael Biebl [2016-02-01 16:27 -0800]:

> Dropping the concept of a wheel aka. sudo group won't work

Why wouldn't it? Especially since we've just been dropping the "wheel"
part of the ACLs forever in Debian :-) There is already the "adm"
group which is meant for log reading.

For the record, I'm not sure that we can/should just replace "wheel"
with "sudo" -- sudo might not even be installed on a machine.

**Comment by johannbg** [comment]
Since things are heading down this road you are looking at a switch for every user and group that is not strictly required and the name of those switches should always match the entry as defined by configuration file(s) that reside in sysusers.d. directory.

What I'm trying to understand here is given that @poettering has accepted the fact systemd is shipping "bad" upstream defaults by allowing removal of the wheel group he can just as well drop all the legacy bit and expect downstream distribution to configure/enable/ship that, rename basic.conf to systemd.conf and reduce the entry's there to strict system(d) functional requirements ( which probably is just root and kernel and userspace facilities ) and expect downstream distributions and consumer to add everything else ( sysusers.conf or distro,conf etc )  and skip adding or asking for "switch" patches. 

You either set the bases which everyone follows no exception ( or else you fall into the rabbit whole chasing every deviation out there and the code mess that follows that ) or you reduce the upstream defaults to strict functional requirement and expect everyone else implement their own deviation from that downstream.

**Comment by poettering** [comment]
I figure this can be closed now that #2621 got merged.
